Enhanced Patch9Frame

Geequlim opened the pull request:
Here are a lot of changes:
- Add texture region support for Patch9Frame
- Add texture region support for stylebox rendering which can render the new Patch9Frame
- Add a plugin to quick select texture region and patch margins for Patch9Frame.

The GIF screen recorder:
![](https://cdn.rawgit.com/Geequlim/depot/master/images/godot/patch9plugin.gif)

Geequlim committed:
Add texture region support for stylebox render

Geequlim committed:
Add texture region support for Patch9Frame

Geequlim committed:
Add plugin to pick texture region and patch margins for Patch9Frame

reduz commented:
Seems OK in general, but wouldn't it be better to just modify the
SpriteRgionEditorPlugin to also work with this?

On Fri, Jun 3, 2016 at 5:22 PM, Geequlim <notifications@github.com> wrote:

> Here are a lot of changes here.
>
>    - Add texture region support for Patch9Frame
>    - Add texture region support for stylebox rendering which can render
>    the new Patch9Frame
>    - Add a plugin to quick select texture region and patch margins for
>    Patch9Frame.
>
> The GIF screen recorder:
>
> <https://camo.githubusercontent.com/692e6043aaed12fe1fee6b076f24aef01b179a7c/68747470733a2f2f63646e2e7261776769742e636f6d2f47656571756c696d2f6465706f742f6d61737465722f696d616765732f676f646f742f706174636839706c7567696e2e676966>
> ------------------------------
> You can view, comment on, or merge this pull request online at:
>
>   https://github.com/godotengine/godot/pull/5026
> Commit Summary
>
>    - Add texture region support for stylebox render
>    - Add texture region support for Patch9Frame
>    - Add plugin to pick texture region and patch margins for Patch9Frame
>
> File Changes
>
>    - *M* core/command_queue_mt.h
>    <https://github.com/godotengine/godot/pull/5026/files#diff-0> (126)
>    - *M* drivers/gles2/rasterizer_gles2.cpp
>    <https://github.com/godotengine/godot/pull/5026/files#diff-1> (45)
>    - *M* drivers/gles2/rasterizer_gles2.h
>    <https://github.com/godotengine/godot/pull/5026/files#diff-2> (2)
>    - *M* platform/iphone/rasterizer_iphone.cpp
>    <https://github.com/godotengine/godot/pull/5026/files#diff-3> (26)
>    - *M* scene/gui/patch_9_frame.cpp
>    <https://github.com/godotengine/godot/pull/5026/files#diff-4> (38)
>    - *M* scene/gui/patch_9_frame.h
>    <https://github.com/godotengine/godot/pull/5026/files#diff-5> (4)
>    - *M* scene/resources/style_box.cpp
>    <https://github.com/godotengine/godot/pull/5026/files#diff-6> (3)
>    - *M* servers/server_wrap_mt_common.h
>    <https://github.com/godotengine/godot/pull/5026/files#diff-7> (61)
>    - *M* servers/visual/rasterizer.h
>    <https://github.com/godotengine/godot/pull/5026/files#diff-8> (5)
>    - *M* servers/visual/rasterizer_dummy.cpp
>    <https://github.com/godotengine/godot/pull/5026/files#diff-9> (3)
>    - *M* servers/visual/rasterizer_dummy.h
>    <https://github.com/godotengine/godot/pull/5026/files#diff-10> (2)
>    - *M* servers/visual/visual_server_raster.cpp
>    <https://github.com/godotengine/godot/pull/5026/files#diff-11> (3)
>    - *M* servers/visual/visual_server_raster.h
>    <https://github.com/godotengine/godot/pull/5026/files#diff-12> (2)
>    - *M* servers/visual/visual_server_wrap_mt.h
>    <https://github.com/godotengine/godot/pull/5026/files#diff-13> (2)
>    - *M* servers/visual_server.cpp
>    <https://github.com/godotengine/godot/pull/5026/files#diff-14> (6)
>    - *M* servers/visual_server.h
>    <https://github.com/godotengine/godot/pull/5026/files#diff-15> (4)
>    - *M* tools/editor/editor_node.cpp
>    <https://github.com/godotengine/godot/pull/5026/files#diff-16> (7)
>    - *A* tools/editor/plugins/patch9frame_region_editor_plugin.cpp
>    <https://github.com/godotengine/godot/pull/5026/files#diff-17> (648)
>    - *A* tools/editor/plugins/patch9frame_region_editor_plugin.h
>    <https://github.com/godotengine/godot/pull/5026/files#diff-18> (140)
>
> Patch Links:
>
>    - https://github.com/godotengine/godot/pull/5026.patch
>    - https://github.com/godotengine/godot/pull/5026.diff
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/godotengine/godot/pull/5026>, or mute the thread
> <https://github.com/notifications/unsubscribe/AF-Z216wOo6xdJ1fzBWDtXv3Jp-2Quk-ks5qII0IgaJpZM4It4EP>
> .
>

Geequlim commented:
@reduz That's good ideal. 
I think that would change the meaning of `SpriteRgionEditorPlugin` so we should give it a new name.

reduz commented:
you could rename it to TextureRegionEditorPlugin


On Fri, Jun 3, 2016 at 5:35 PM, Geequlim <notifications@github.com> wrote:

> @reduz <https://github.com/reduz> That's good ideal.
> I think that would change the meaning of SpriteRgionEditorPlugin so we
> should give it a new name.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/godotengine/godot/pull/5026#issuecomment-223687856>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AF-Z2-pl_QMrRvPzhXL5N1zJmBzPr-elks5qIJArgaJpZM4It4EP>
> .
>

Geequlim commented:
Now SpriteRegionEditorPlugin and Patch9FrameRegionEditorPlugin are merged into TextureRegionEditorPlugin it works for Sprite and Patch9Frame.

Geequlim commented:
Close #4842 when merged :)

akien-mga commented on scene/gui/patch_9_frame.cpp:
```diff
@@ -95,6 +95,20 @@ void Patch9Frame::set_patch_margin(Margin p_margin,int p_size) {
 	margin[p_margin]=p_size;
 	update();
 	minimum_size_changed();
+    switch (p_margin) {
```
Looks like you used spaces for indentation instead of tabs, could you have a look at fixing that?

akien-mga commented:
Looks like most changes use spaces instead of tabs for indentation actually, it would be nice if you could amend the commits to fix it.
The last two commits should probably be squashed together too.

Geequlim commented:
@akien-mga Is there any coding style statement for the C++ code?
I hope something like [EditorConfig](http://editorconfig.org/) to tell contributors how to format the code.

akien-mga commented:
@Geequlim Not yet, but this is being discussed in #3916. Please mention EditorConfig there, it sounds like a good enhancement.

Basically the current coding style standard is to use the same standard as the rest of the code :)

Geequlim commented:
@akien-mga Well, I will reformat the code and squash commits :)

Geequlim commented:
@akien-mga I have replaced space indents to tabs for new file.
If anything should to do with this pr just tell me :)

Geequlim committed:
Enhanced StyleBoxTexture:
  Add texture region support for StyleBoxTexture.
  Quick region(texture region and scale region) selection for StyleBoxTexture with TextureRegionEditorPlugin.

Geequlim commented:
Now the TextureRegionEditorPlugin works for StyleBoxTexture resources.
Patch9 images can be easily used with various controls.
